### PR TITLE
Monitoring Module

### LINKED ISSUE #88: Monitoring module
`cardano.monitoring.enable = true`

- prometheus
- grafana
- prometheus exporter configuration
- networking configuration
- acme
- example
- test
- documentation

### PR DESCRIPTION
**Pull request by brainrake**
- [x] `cardano.monitoring` nixos module
- [x] prometheus exporter configuration
- [x] prometheus server
- [x] prometheus scraper configuration
- [x] grafana server
- [x] networking and firewall configuration
- [x] test
- [x] example
- [x] documentation
- [x] more doc
- [x] cloud deployment documentation: split from load balancer documentation 
- [x] template: rename (multi-host, full, or something)
- [x] fix ports used for firewall, use service configs
- prometheus exporters
  - [x] node (host)
  - [x] nginx
  - [x] cardano-node
  - [x] postgres
  - [x] ogmios
  - [x] kupo
  - [x] blockfrost
  - [x] db-sync
  - [x] oura
- prometheus scrapers
  - [x] node (host)
  - [x] nginx
  - [x] postgres
  - [x] cardano-node
  - [x] ogmios
  - [x] kupo
  - [x] blockfrost
  - [x] db-sync
  - [x] oura
- grafana dashboards
  - [x] node (host)
  - [x] nginx
  - [x] postgres
  - ~cardano-node~
  - ~ogmios~
  - ~kupo~
  - ~blockfrost~
  - ~db-sync~ 
  - ~oura~
- [x] update nixpkgs to fix vm issue
- [x] fix doc rendering

as i couldn't find working grafana dashboards for all the services, we decided on the 2024-10-01 meeting to build a basic dashboard for cardano-node in #109  and consider dashboards for the other services out of scope
 
closes #88

### CODE REVIEW

File: modules/monitoring.nix [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      ports = mkOption {
+        type = with types; listOf port;
+        default = [
+          config.services.blockfrost.settings.server.port
+          config.services.cardano-db-sync.explorerConfig.PrometheusPort
+          config.services.ogmios.port
+          config.services.prometheus.exporters.node.port
+          config.services.prometheus.exporters.nginx.port
+          config.services.prometheus.exporters.postgres.port
+          12798 # cardano-node
```
Comment (avnik) [inline modules/monitoring.nix:38]: Could we avoid magic numbers here?

**Review by avnik — CHANGES_REQUESTED** [review]
Looks sane for me. As I see  is also have some Loadbalancer related changes.
Could you please address a magic numbers which I denote in code?

File: modules/monitoring.nix (hunk 1 above)
Comment (brainrake) [reply modules/monitoring.nix]: Added options.

**Review by avnik — APPROVED** [review]
LGTM

**Pull request merged**